### PR TITLE
added option for fire toggle

### PR DESCRIPTION
**Pull request by aviralch**

### CODE REVIEW

**Comment by MahirJhaveri** [comment]
Whats the status?

**Comment by aviralch** [comment]
Midterm at 10 am, project on Monday and Wednesday. Will work on it on Thursday.

**Comment by MahirJhaveri** [comment]
A quick overview of this task (as we spoke last weekend):

You will need to:

- Update both the client and the server (since atm bullets are created by the server every X seconds, which should not be the case. So simply making the bullets invisible won't help)

- Add a new keypress event which sends a new (WebSocket) message to the server notifying of the keypress (look how the direction keys are handled for this)

- Now when autofire is enabled, let the server do what it is currently doing (as of now the bullet creation happens in update() function of player.js). But when autofire is disabled you want the server to only create new bullets when the keypress even is received, also need to make sure that you are waiting for the cooldown period to end before creating a new bullet (to control how fast the user is firing). For this you'll most likely have to add another function in player.js (something like fireBullet()) which is called everytime the key is pressed and it should create the bullet and update the related game state.

Files to look out for (only mentioning the ones on the server-side, assuming you are familiar frontend):

- player.js

- game.js

**Comment by MahirJhaveri** [comment]
status??

**Comment by aviralch** [comment]
Having fun with understanding JS Hoisting

**Comment by aviralch** [comment]
There seems to be one issue left, when one selects auto toggle as off, bullets seem to pile up. So like if i say toggle is off. I won't shoot bullets (say 10) but as soon as I click all 10 bullets are shot at once.

**Comment by aviralch** [comment]
added a threshold value for negative value of fireCooldown to prevent one shot kill

**Comment by aviralch** [comment]
Hey, any updates yet ?

**Comment by MahirJhaveri** [comment]
Did see the PR. A couple of things:

1) GREAT WORK!! 🚀
2) The switch is labeled wrong. It is auto-firing on manual fire setting and manual firing on the autofire setting.
3) Try to get this in sync with the current master and then we will do one final test and then get it merged!!

File: src/server/server.js [hunk 1]
```diff
@@ -42,7 +44,9 @@ const game = new Game();
 // NOTE: This keyword ??? idk how that works??
 
 function joinGame(msg) {
-    game.addPlayer(this, msg.username, (msg.color) ? msg.color : Constants.TANK.BLUE);
+    console.log("HERE");
```
Comment (MahirJhaveri) [inline src/server/server.js:47]: please remove all these logs

File: src/server/server.js [hunk 2]
```diff
@@ -42,7 +44,9 @@ const game = new Game();
...
+    console.log(msg);
```
Comment (MahirJhaveri) [inline src/server/server.js:48]: please remove this also

Comment (aviralch) [reply src/server/server.js]: Corrected in the latest commit

**Comment by aviralch** [comment]
Approve status ?

**Comment by MahirJhaveri** [comment]
LGTM!! Merging rn

**Review by MahirJhaveri — APPROVED** [review]

**Pull request merged**